clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java

```diff
@@ -159,24 +159,25 @@ private static ChannelBuilder create(SecurityProtocol securityProtocol,
     }
 
     // Visibility for testing
-    protected static Map<String, Object> channelBuilderConfigs(final AbstractConfig config, final ListenerName listenerName) {
-        Map<String, ?> parsedConfigs;
+    @SuppressWarnings("unchecked")
+    static Map<String, Object> channelBuilderConfigs(final AbstractConfig config, final ListenerName listenerName) {
+        Map<String, Object> parsedConfigs;
         if (listenerName == null)
-            parsedConfigs = config.values();
+            parsedConfigs = (Map<String, Object>) config.values();
```

Comment:
Does this cover the case when listenerName is not null? I guess that can only happen on the server side and since we don't log unused configs on the server, so maybe this is ok for now?

Reply:
```java
        if (listenerName == null)
            parsedConfigs = (Map<String, Object>) config.values();
        else
            parsedConfigs = config.valuesWithPrefixOverride(listenerName.configPrefix());
```

the method ```config.valuesWithPrefixOverride``` also returns ```RecordingMap so it is ok.